azure/services/publicips/spec.go

```diff
@@ -59,44 +59,44 @@ func (s *PublicIPSpec) OwnerResourceName() string {
 // Parameters returns the parameters for the public IP.
 func (s *PublicIPSpec) Parameters(ctx context.Context, existing interface{}) (params interface{}, err error) {
 	if existing != nil {
-		if _, ok := existing.(network.PublicIPAddress); !ok {
-			return nil, errors.Errorf("%T is not a network.PublicIPAddress", existing)
+		if _, ok := existing.(armnetwork.PublicIPAddress); !ok {
+			return nil, errors.Errorf("%T is not a armnetwork.PublicIPAddress", existing)
 		}
 		// public IP already exists
 		return nil, nil
 	}
 
-	addressVersion := network.IPVersionIPv4
+	addressVersion := armnetwork.IPVersionIPv4
 	if s.IsIPv6 {
-		addressVersion = network.IPVersionIPv6
+		addressVersion = armnetwork.IPVersionIPv6
 	}
 
 	// only set DNS properties if there is a DNS name specified
-	var dnsSettings *network.PublicIPAddressDNSSettings
+	var dnsSettings *armnetwork.PublicIPAddressDNSSettings
 	if s.DNSName != "" {
-		dnsSettings = &network.PublicIPAddressDNSSettings{
+		dnsSettings = &armnetwork.PublicIPAddressDNSSettings{
 			DomainNameLabel: ptr.To(strings.Split(s.DNSName, ".")[0]),
 			Fqdn:            ptr.To(s.DNSName),
 		}
 	}
 
-	return network.PublicIPAddress{
+	return armnetwork.PublicIPAddress{
 		Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
 			ClusterName: s.ClusterName,
 			Lifecycle:   infrav1.ResourceLifecycleOwned,
 			Name:        ptr.To(s.Name),
 			Additional:  s.AdditionalTags,
 		})),
-		Sku:              &network.PublicIPAddressSku{Name: network.PublicIPAddressSkuNameStandard},
+		SKU:              &armnetwork.PublicIPAddressSKU{Name: ptr.To(armnetwork.PublicIPAddressSKUNameStandard)},
 		Name:             ptr.To(s.Name),
 		Location:         ptr.To(s.Location),
-		ExtendedLocation: converters.ExtendedLocationToNetworkSDK(s.ExtendedLocation),
-		PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
-			PublicIPAddressVersion:   addressVersion,
-			PublicIPAllocationMethod: network.IPAllocationMethodStatic,
+		ExtendedLocation: converters.ExtendedLocationToNetworkSDKv2(s.ExtendedLocation),
```

Comment:
It looks like this was the last remaining instance of the "v1" version of this function, so we can probably remove that now and rename the v2 one to drop the "v2"

Reply:
Updated via 7ba1619f6df44b7cc1ffb572c216a18297a7c4ed